### PR TITLE
Making watch work with read_namespaced_pod_log

### PR DESCRIPTION
**Pull request by mitar**
Fixes https://github.com/kubernetes-client/python/issues/199.

### CODE REVIEW

**Comment by yliaog** [comment]
please add a test case for this.

File: watch/watch.py [hunk 1]
```diff
@@ -132,9 +142,12 @@ def stream(self, func, *args, **kwargs):
                     if self._stop:
                         break
             finally:
-                kwargs['resource_version'] = self.resource_version
                 resp.close()
                 resp.release_conn()
+                if self.resource_version is not None:
+                    kwargs['resource_version'] = self.resource_version
+                else:
```
Comment (micw523) [inline watch/watch.py]: Do you need the else: break here?

Comment (mitar) [reply watch/watch.py]: Yes. The log streaming does not have `resource_version`. So once the response finishes, you have to stop.

**Comment by mitar** [comment]
@yliaog I have added tests.

@micw523 Please review and maybe merge this?

**Comment by micw523** [comment]
As much as I want to merge this I don't have ownership of this repo :)
/lgtm
/cc @roycaihw

File: watch/watch.py (hunk 1 above)
Comment (mitar) [reply watch/watch.py]: In fact that was not good because using break inside finally swallows up the exception. I change it to setting a flag.

**Pull request merged**